libV2/schemaUtils.js

```diff
@@ -0,0 +1,1603 @@
+    };
+  },
+
+  resolveResponseHeaders = (context, responseHeaders) => {
+    const headers = [];
+
+    if (_.has(responseHeaders, '$ref')) {
+      responseHeaders = resolveSchema(context, responseHeaders);
+    }
+
+    _.forOwn(responseHeaders, (value, headerName) => {
+      let headerValue = resolveValueOfParameter(context, value);
+
+      if (typeof headerValue === 'number' || typeof headerValue === 'boolean') {
+        // the SDK will keep the number-ness,
+        // which will be rejected by the collection v2 schema
+        // converting to string to prevent issues like
+        // https://github.com/postmanlabs/postman-app-support/issues/6500
+        headerValue = headerValue.toString();
+      }
+
+      const headerData = Object.assign({}, value, { name: headerName }),
+        serialisedHeader = serialiseParamsBasedOnStyle(context, headerData, headerValue);
+
+      headers.push(...serialisedHeader);
+    });
+
+    return headers;
+  },
+
+  getPreviewLangugaForResponseBody = (bodyType) => {
+    const headerFamily = getHeaderFamily(bodyType);
+
+    return HEADER_TYPE_PREVIEW_LANGUAGE_MAP[headerFamily] || 'text';
+  },
+
+  resolveResponseForPostmanRequest = (context, operationItem, originalRequest) => {
+    let responses = [];
+
+    _.forOwn(operationItem.responses, (responseSchema, code) => {
+      let response,
+        { body, contentHeader = [], bodyType } = resolveResponseBody(context, responseSchema) || {},
+        headers = resolveResponseHeaders(context, responseSchema.headers);
+
+      response = {
+        name: _.get(responseSchema, 'description'),
+        body,
+        headers: _.concat(contentHeader, headers),
+        code,
+        originalRequest,
+        _postman_previewlanguage: getPreviewLangugaForResponseBody(bodyType)
+      };
+
+      responses.push(response);
+    });
+
+    return responses;
+  };
+
+module.exports = {
+  resolvePostmanRequest: function (context, operationItem, path, method) {
+    /**
+     * schemaCache object will be used to cache the already resolved refs
+     * in the schema.
+     */
+    context.schemaCache = context.schemaCache || {};
+    context.schemaFakerCache = context.schemaFakerCache || {};
+
+    let url = resolveUrlForPostmanRequest(path),
+      baseUrlData = resolveBaseUrlForPostmanRequest(operationItem[method]),
+      requestName = resolveNameForPostmanReqeust(context, operationItem[method], url),
+      queryParams = resolveQueryParamsForPostmanRequest(context, operationItem, method),
+      headers = resolveHeadersForPostmanRequest(context, operationItem, method),
+      pathParams = resolvePathParamsForPostmanRequest(context, operationItem, method),
+      { pathVariables, collectionVariables } = filterCollectionAndPathVariables(url, pathParams),
+      requestBody = resolveRequestBodyForPostmanRequest(context, operationItem[method]),
+      request,
+      responses,
+      securitySchema = _.get(operationItem, [method, 'properties.security']),
```

Comment:
@barshan23 Why are we accessing `properties` here? There should be no such properties defined for OAS security object.